src/Analyser/NodeScopeResolver.php

```diff
@@ -1465,6 +1465,24 @@ function (MutatingScope $scope) use ($expr, $nodeCallback, $context): Expression
 			) {
 				$scope = $scope->assignVariable('http_response_header', new ArrayType(new IntegerType(), new StringType()));
 			}
+
+			if (
+				isset($functionReflection)
+				&& $functionReflection->getName() === 'extract'
+				&& $scope->getType($expr->args[0]->value) instanceof ConstantArrayType
```

Comment:
You're accessing `args[0]` before checking that the argument was passed to the function call.

Reply:
This means that analyzing code with `extract()` will fail with an internal error. Which shouldn't be the case.